Source/Extensions/UIViewExtension.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  UITapGestureRecognizerExtension.swift
+//  Frames
+//
+//  Created by Ehab Alsharkawy
+//  Copyright © 2022 Checkout. All rights reserved.
+//
+
+import UIKit
+
+extension UIView {
+
+  static var keyboardDismissTapGesture: UITapGestureRecognizer {
+    let gesture = UITapGestureRecognizer(target: self, action: #selector(hideKeyboard(gestureRecognizer:)))
+    gesture.cancelsTouchesInView = false
+    return gesture
+  }
+
+  @objc private static func hideKeyboard(gestureRecognizer: UITapGestureRecognizer) {
+    let view = gestureRecognizer.view
+    let loc = gestureRecognizer.location(in: view)
+    let subview = view?.hitTest(loc, with: nil)
+    guard !(subview is SecureDisplayView) else { return }
+    UIApplication.shared.sendAction(#selector(UIApplication.resignFirstResponder), to: nil, from: nil, for: nil)
```

Comment:
1. Do we need `UIApplication.shared.sendAction(#selector(UIApplication.resignFirstResponder), to: nil, from: nil, for: nil)`, I thought `UIApplication.shared.resignFirstResponder()` might be sufficient?

2. Do we need to check for hit test? I've previously asked whether that view catches the gesture already. Sorry but haven't seen any reply and I still think it would 🤔

Reply:
UIApplication.shared.resignFirstResponder() is not working

Reply:
`resignFirstResponder()` was my mistake sorry. I've reviewed the behaviour and it is only working via the `sendAction` dispatch.

I've also tried locally to drop the hitTest and I can see this is a good approach in this situation 👍 . Consider this thread resolved from my side